components/donateTicketForm.js

```diff
@@ -0,0 +1,231 @@
+/* global fetch */
+import React, { Component } from 'react'
+import { CardElement } from '@stripe/react-stripe-js'
+import DonationFrequency from './donationFrequency'
+import Router from 'next/router'
+import { Input, FormControl, FormErrorMessage, InputGroup, InputLeftElement } from '@chakra-ui/react'
+import { Form, Formik } from 'formik'
+import { validateEmail, validateText } from '../utils/validation.util'
+import { fundraisingEventTicket, fundraisingEventTicketPatron } from '../lib/constants'
+
+class DonateTicketForm extends Component {
+  constructor (props) {
+    super(props)
+    this.donate = this.donate.bind(this)
+    this.frequenciesAvailable = [fundraisingEventTicket, fundraisingEventTicketPatron]
+    this.state = {
+      loading: false,
+      redirectSuccess: false,
+      previousFrequency: fundraisingEventTicket
+    }
+  }
+
+  setLocalState = (state) => {
+    if (!state.error) state.error = ''
+    this.setState(state)
+  }
+
+  customOnFrequencyChange = (e, handleChange, setFieldValue) => {
+    const frequency = e.currentTarget.value
+    if (frequency !== this.state.previousFrequency) {
+      this.state.previousFrequency = frequency
+      if (frequency === fundraisingEventTicket) {
+        setFieldValue('amount', '100')
+      } else if (frequency === fundraisingEventTicketPatron) {
+        setFieldValue('amount', '200')
+      }
+    }
+    handleChange(e)
+  }
+
+  donate = async (formValues) => {
+    this.setLocalState({ loading: true })
+    let token
+    try {
+      const cardElement = this.props.elements.getElement(CardElement)
+      const res = await this.props.stripe.createToken(cardElement)
+      token = res.token
+    } catch (e) {
+      this.setState({ error: e.message, loading: false })
+      return
+    }
+
+    if (!token) {
+      this.setState({ error: 'Invalid CC info!', loading: false })
+      return
+    }
+    try {
+      const { frequency, firstName, lastName, email, amount } = formValues
+      const responseStream = await fetch('/api/purchase-ticket', {
+        method: 'POST',
+        body: JSON.stringify({
+          source: token,
+          firstName,
+          frequency,
+          lastName,
+          amount: amount * 100,
+          email
+        })
+      })
+      const response = await responseStream.json()
+      if (response.success) {
+        this.setState({ redirectSuccess: true, loading: false })
+      } else {
+        this.setState({ error: `Donation failed: ${response.message}`, loading: false })
```

Comment:
Probs I'd not expose the msg directly just in case, and instead direct users to an alternate method. Maybe email x@theteacherfund.com to get your ticket?